tests/tests.py

```diff
@@ -890,3 +893,98 @@ def test_invalidate_model_with_geometry(self):
         geom.save()
         # Raises ValueError if this doesn't work
         invalidate_obj(geom)
+
+
+class MultiDBInvalidationTests(BaseTestCase):
+
+    @override_settings(CACHEOPS_PREFIX=lambda q: q.db)
```

Comment:
Should not be needed since this model is `'db_agnostic': False`

Reply:
That's why I used that class but I put that decorator because otherwise it doesn't work. If I correctly understand your code this is the reason : when you cache a queryset, the db_agnostic settings implies (as we wish) that cache key are different for the 2 query set because they use a different db connection. However, their prefix is the same because default CACHEOPS_PREFIX  returns an empty string. The problem is that when you invalidate the cache (in bulk_create for instance), the invalidation for each object is based only on the prefix, db_table and obj_value. Three are the same for 2 identical queries on different DB so no DB distinction is done during invalidation. If my understanding is correct, a way to keep in line with the current behavior is perhaps to use a default CACHEOPS_PREFIX which return empty string if a model is not db_agnostic and the db connection name if it is. If so, the db_agnostic clause in _cache_key method would be removed.